Khan/wonder-blocks
# Clickable components no longer click when being dragged into

## Summary:
Currently, Clickable components are activated on mouse release, when
they should only be activated on click.

Changes
- Removed the redundant `this.handleClick` from `handleMouseEnter`
  in clickable-behavior.js.
- After that step, the dragging logic doesn't actually do anything
  anymore, so I removed that too.
- Made it so that mouseenter only affects the hover state, not pressed.

Issue: https://khanacademy.atlassian.net/browse/WB-1425

## Test plan:
- Open http://localhost:6061/?path=/docs/clickable-clickable--default
- Click the clickable element. Confirm that the click alert pops up.
- Press inside the clickable element, drag out, and unpress. Confirm it does NOT result in a click alert.
- Press outside the clickable element, drag in, and unpress. Confirm it does NOT result in a click alert.
- Press inside the clickable element, drag within it, and unpress. Confirm that the click alert pops up.
- Press inside the clickable element, drag outside it, DON'T unpress, drag back into it, and then unpress.
  Confirm that the click alert pops up.
---
- Go to mobile storybook locally.
- Confirm that tapping clicks the button.
- Confirm that dragging into the button does not click it.
- Confirm that dragging out of the button does not click it.
- Dragging within and the button and out/back in does not apply in mobile.
---
`yarn jest packages/wonder-blocks-clickable/src/components/__tests__/clickable-behavior.test.js`

## Before:
(Hovering and then dragging)

https://user-images.githubusercontent.com/13231763/210906862-2c54fb4b-6503-48e0-ade3-e01a7550e28e.mov

## After:
(Hovering, dragging, and clicking)

https://user-images.githubusercontent.com/13231763/210906877-158ffcc1-bb4a-443f-ad77-0bb5b25d9eae.mov
